packages/gatsby-theme-docs/src/components/link.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Location } from '@reach/router';
+import { Link as GatsbyLink, withPrefix } from 'gatsby';
+import { css } from '@emotion/core';
+import styled from '@emotion/styled';
+import { OutboundLink } from 'gatsby-plugin-google-analytics';
+import ExternalLinkIcon from '../icons/external-link-icon.svg';
+import { colors, dimensions } from '../design-system';
+import { useSiteData } from '../hooks/use-site-data';
+
+const isProduction = process.env.NODE_ENV === 'production';
+
+const productionHostname = 'docs.commercetools.com';
+const dummyHostname = 'dummy.com';
+
+// Gatsby links should be written with the full url path starting with `/` (so no
+// "filesystem"-relative paths).
+// In order to resolve relative paths that do no start with a `/`, we need to first
+// trim the possible `pathPrefix` included in the `location` of `@react/router`.
+const withoutPrefix = (value, pathPrefix) =>
+  value.replace(new RegExp(`^${pathPrefix}`), '');
+
+const linkStyles = css`
+  text-decoration: underline;
+  &,
+  > code {
+    color: ${colors.light.link};
+    :active,
+    :focus,
+    :hover {
+      color: ${colors.light.linkHover};
+    }
+  }
+`;
+
+export const ExternalSiteLink = props => (
+  <OutboundLink
+    {...props}
+    css={linkStyles}
+    target="_blank"
+    rel="noopener noreferrer"
+  />
+);
+const AnchorLink = styled.a`
+  ${linkStyles}
+`;
+const InternalSiteLink = styled(AnchorLink)``;
+const HistoryLink = styled(GatsbyLink)`
```

Comment:
I named those components after their contextual usage:
- `ExternalSiteLink`
- `InternalSiteLink`
- `AnchorLink`
- `HistoryLink`

Reply:
to be honest all three name choices (InternalSiteLink, HistoryLink and InlineLink) do not feel logical to me.   

"InternalSite" is just for "on the same page" but not for "on the same site".  

"History" is a different concern to me than the linking itself.  You can choose to make  a GatsbyLink triggering a history or not but that's a different angle.   

And "Inline" rather sounds like inline vs. block to me but in the implementation it is used for external links.

Reply:
Inline is not one of the 4 main components listed above, it’s just for styling.

We can improve the naming if you want, the point is that we have those 4 different link types 😉

Reply:
OK, i suggest to think about a better name for `HistoryLink` only.  I would prefer to stick with GatsbyLink becuase it's key property is that we delegate the handling to gatsbyJS to do whatever it thinks it should do with it.   If you want it less implementation-specific it's  a sort of  In-App Link.  

Other links also write history and the GatsbyLink can choose to not write history depending on its configuration and parameters, that's why I am not convinced of "History".

Reply:
Technically "history" is the correct term, as it refers to the [browser history](https://developer.mozilla.org/en-US/docs/Web/API/History) which is what Gatsby uses through `@reach/router`.

My intent with the naming was about that and making it clear how the link behaves. If you prefer to keep GatsbyLink it's fine, I don't want to argue much about the naming.
I just wanted to clarify the intention behind those choices from a technical perspective.